src/10-ipfs-0-4-0-released/index.md

```diff
@@ -0,0 +1,159 @@
+---
+date: 2016-03-17
+id: 10-ipfs-0-4-0-released
+template: tmpl/layouts/post.html
+baseurl: ..
+breadcrumbs:
+- {name: "10-ipfs-0-4-0-released", link: "./" }
+tags: modules
+title: IPFS 0.4.0 has been released
+author: Kyle Drake and Whyrusleeping
+collection: posts
+---
+
+[IPFS 0.4.0](https://github.com/ipfs/go-ipfs/releases/tag/v0.4.0) has been
+released! Among the many changes are a revamped implementation of the IPFS
+communication protocols, increased performance, improvements to IPNS (The
+Interplanetary Naming System), many bugfixes, and lots of new features to make
+IPFS even more powerful.
+
+## Why we're changing the protocol
+
+The most important change allows IPFS implementations to use pluggable
+stream multiplexers, such as [yamux](https://github.com/hashicorp/yamux),
+[spdystream](https://github.com/docker/spdystream), or
+[muxado](https://github.com/inconshreveable/muxado). Instead of locking IPFS
+permanently into a single multiplexer that won't work for every language or
+situation, this change allows each IPFS implementation to implement the multiplexers
+of their choosing, and then negotiate which to use during the initial connection
+handshake.
+
+This modularity with stream muxing makes it easier for certain languages to
+improve performance. For example, the Go programming language may have muxado
+and yamux implementations that are really good, but many languages lack good (or
+any) implementations of them. Or take Node.js, which works well with spdystream
+in testing, and it would be nice to take advantage of that. And then there are
+options like ['multiplex'](https://github.com/maxogden/multiplex), which may
+not have the same performance, but are much easier to implement.
+
+So, by supporting as many muxers as we can, we get to choose the best
+multiplexers for the job. It also makes it much easier to implement the IPFS
+protocols in a new language. And of course, if a better multiplexers standard
+comes along, it will be easier to upgrade IPFS to support it in the future.
+
+The same code that allows us to easily select a stream muxer is also being
+used to select which IPFS sub-protocol to use for any given stream between
+peers. Now, if we need to make a breaking protocol change to any of those (like
+the DHT or bitswap) we can do so seamlessly, and provide easy backwards
+compatibility. We won't need to "break" anything because we will be able to 
+have nodes run multiple protocols at the same time for compatibility.
+
+In addition to the multiplexers changes, the protocol revamp has also improved
+efficiency and performance in a few important ways, including the elimination of
+a double wrapping of the length prefixer, and the removal of some unnecessary
+round trips between nodes.
+
+### Breaking changes
+
+This is a **breaking change** release, with a major refactor and upgrade to the
```

Comment:
I'm afraid that `**breaking change**` alone might mislead people in thinking that we broke data linking. Calling it a `network breaking change` and then explaining in the end of the paragraph that data remains intact will avoid many doubts